src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java

```diff
@@ -181,8 +181,8 @@ private static boolean isEmptyTag(DetailNode tagNode) {
     private static boolean hasOnlyEmptyText(DetailNode description) {
         boolean result = true;
         for (DetailNode child : description.getChildren()) {
-            if (child.getType() != JavadocTokenTypes.TEXT
-                    || !CommonUtil.isBlank(child.getText())) {
+            if (child.getType() != JavadocTokenTypes.LEADING_ASTERISK
+                    && !CommonUtil.isBlank(child.getText())) {
```

Comment:
See https://github.com/checkstyle/checkstyle/pull/12979#issuecomment-1520460709, unless we have exception in check, open a new issue using the snapshot from this PR and revert changes to this check.

Reply:
@nrmancuso The problem was that the grammar and check are related to each other cause of "branch coverage".
In old grammar, `@return` description is mandatory, so in any case it will take the next statement and description will never be `null`, so `|| hasOnlyEmptyText(tagDescription);` will be executed.

In new grammar, description can be `null`, which means if the description was empty `return tagDescription == null` will be executed and we will never need to call `hasOnlyEmptyText()`.

That's why I thought we need to update both in one -this- PR.